mongoose/mongoose.d.ts

```diff
@@ -871,7 +871,7 @@ declare module "mongoose" {
     /** Hash containing current validation errors. */
     errors: Object;
     /** The string version of this documents _id. */
-    id: string;
+    id?: string;
```

Comment:
Doing so breaks
````
class Subdocument extends MongooseDocument
````
and
````
class Embedded extends MongooseDocument
````

Do not have enough insights to judge if those can become interfaces as well, and if that has further implications throughout the declaration.